inference-engine/src/inference_engine/ie_compound_blob.cpp

```diff
@@ -192,6 +192,50 @@ void verifyI420BlobInput(const Blob::Ptr& y, const Blob::Ptr& u, const Blob::Ptr
     }
 }
 
+TensorDesc verifyBatchedBlobInput(const std::vector<Blob::Ptr>& blobs) {
+    // verify invariants
+    if (blobs.size() < 1) {
+        THROW_IE_EXCEPTION << "at least 1 sub-blob needed";
+    }
+
```

Comment:
@myshevts should not we check that all Blob have the same type? It can simplify new mode for MULTI `DATA_AFFINITY_MODE`.. in this case we will not have ambiguates like BatchedBlob contains blobs for KMB & GPU

Reply:
The code should test that all blobs are from the same device (==device where the InferRequest is originated via ExecutableNetwork), regardless of MULTI, so this is good point

Reply:
It is not necessary to use only one kind of blobs here if if plugin which receives the BatchedBlob can handle each of them correctly - e.g. use `XXmap()` for unknown types.

Reply:
@ilya-lavrenov , @myshevts can you please response here with a decision: should sub-blobs to be verified for the equivalent types (`typeid()`) or not?

Which other verifications to be done?
- Should `RemoteBlob` to be checked for same `RemoteContext` (leads `RemoteBlob` dependency here)?
- Other checks for other types of blobs?

Thank you.

Reply:
I think we can leave such checks for future changes.

Reply:
ok, thank you